File: lectures/day02/Tesla.java

/**
 * Tesla
 */
public class Tesla {

    private String color;
    private int doors;

    public Tesla(String color, int doors){
        this.doors = doors;
        this.color = color;
    }

    public Tesla(String color){
        this.doors = 4;
        this.color = color;
    }

    public Tesla(){
        this.doors = 4;
        this.color = "black";
    }

    public String getColor() {
        return this.color;
    }

    public void setColor(String color) {
        this.color = color;
    }


    public int getDoors() {
        return this.doors;
    }

    public void setDoors(int doors) {
        this.doors = doors;
    }

    public void Drive(){
        System.out.printf("I am driving my %s Tesla!%n", color);
    }


}

File: lectures/day02/demos/tesla.py

class Tesla:
    
    def __init__(self, color, doors = 4) -> None:
        self.color = color
        self.__doors = doors
        
    def drive(self):
        print(f"I am driving my {self.color}")
        
    # @property
    # def doors(self):
    #     return self.__doors
    
    # @doors.setter
    # def doors(self, doors):
    #     self.__doors = doors